pennylane/ops/op_math/composite.py

```diff
@@ -60,6 +60,7 @@ def __init__(
         self._hash = None
         self._has_overlapping_wires = None
         self._overlapping_ops = None
+        self._pauli_rep = self._build_pauli_rep()
```

Comment:
Just change `build_pauli_rep` by a `_pauli_rep` property, and cache it if needed.